python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py

```diff
@@ -0,0 +1,582 @@
+    """
+    theta = [x[i] for i in range(p)]
+    tau = x[p]
+    # compute conditional mean and standard deviates
+    mean = np.dot(X, theta)
+    std = np.sqrt(1.0 / tau + 1.0 / Q)
+    # inject values in blueprint
+    params = gen_params.copy()
+    params[::4] = mean
+    params[1::4] = std.ravel()
+    return params
+
+
+# %%
+# 2.2.2. Updating :math:`\vect{\theta}`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# :math:`\pi(\vect{\theta}|\vect{Y}, \tau, \vect{Y}^{obs})=\pi(\vect{\theta}|\vect{Y},\tau).` Due to the
+# partial conjugacy of the conditional prior :math:`\pi(\vect{\theta}|\tau),`
+# this is explicit, and given by the following box-constrained
+# multivariate normal:
+#
+# .. math::
+#
+#     \vect{\theta}|\vect{Y},\tau \sim \mathcal N_4( \vect{\mu}_n ; \mat{\Sigma}_n ){\bf 1}_{\{\vect{\theta}_{\min} \leq \vect{\theta} \leq \vect{\theta}_{\max}\}},
+#
+# with
+#
+# .. math::
+#
+#     \vect{\mu}_n = \vect{\mu}_0 + \mat{\Sigma}_0X^\top(\mat{X} \mat{\Sigma}_0 \mat{X}^\top + \mat{I}_n + \tau \mat{Q}^{-1})^{-1}(\vect{Y} - \mat{X} \vect{\mu}_0) \\
+#     \mat{\Sigma}_n = \tau^{-1}\mat{\Sigma}_0 - \tau^{-1}\mat{\Sigma}_0 \mat{X}^\top (\mat{X} \mat{\Sigma}_0 \mat{X}^\top + \mat{I}_n + \tau \mat{Q}^{-1})^{-1} \mat{X} \mat{\Sigma}_0,
+#
+# or equivalently, thanks to the matrix Woodsbury identity:
+#
+# .. math::
+#
+#     \vect{\mu}_n =  (\mat{\Sigma}_0^{-1} + \mat{X}^\top (\mat{I}_n + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}(\mat{\Sigma}_0^{-1} \vect{\mu}_0 + \mat{X}^\top (\mat{I}_n + \tau \mat{Q}^{-1})^{-1} \vect{Y})\\
+#     \mat{\Sigma}_n = \tau^{-1}(\mat{\Sigma}_0^{-1} + \mat{X}^\top (\mat{I}_n + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}.
+#
+# By having all hyperparameters go to :math:`0,` we obtain the following
+# simplified form:
+#
+# .. math::
+#
+#     \vect{\mu}_n =  (\mat{X}^\top (\mat{I}_n + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}(\mat{X}^\top (\mat{I}_n + \tau \mat{Q}^{-1})^{-1} \vect{Y})\\
+#     \mat{\Sigma}_n = \tau^{-1}(\mat{X}^\top (\mat{I}_n + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}.
+#
+# Explicit simulation from a box-constrained multivariate normal can be
+# done with a simple rejection sampling scheme:
+
+# %%
+# Step 1 : Create associated :class:`~openturns.RandomVector`
+
+
+class BoxConstrainedNormal(ot.PythonDistribution):
+    """
+    Multivariate normal distribution
+    under box constraints
+    """
+
+    def __init__(
+        self, d=2, mu=np.zeros(2), Sigma=np.eye(2), r=np.zeros(2), s=np.ones(2)
+    ):
+        for x in mu, Sigma, r, s:
+            if len(x) != d:
+                print("expectation or bound does not have size %s" % d)
+                raise ValueError
+        if Sigma.shape != (d, d):
+            print("covariance matrix not have dimensions (%s,%s)" % (d, d))
+            raise ValueError
+        super(BoxConstrainedNormal, self).__init__(d)
+        self.mu = mu
+        self.Sigma = Sigma
+        self.r = r
+        self.s = s
+
```

Comment:
The line 
```
self.interval = ot.Interval(self.r, self.s)
```
must be inserted there. Otherwise, using `getSample()` fails, because `setParameter()` did not set the `interval` attribute.